logstash-core/lib/logstash/pipeline.rb

```diff
@@ -310,15 +302,37 @@ def output_batch(batch)
 
       # output_func should never return anything other than an Array but we have lots of legacy specs
       # that monkeypatch it and return nil. We can deprecate  "|| []" after fixing these specs
-      outputs_for_event = output_func(event) || []
+      outputs_for_event = with_pipeline_rescue(event) { output_func(event) } || []
 
       outputs_for_event.each { |output| acc[output] << event }
       acc
     end
 
     # Now that we have our output to event mapping we can just invoke each output
     # once with its list of events
-    outputs_events.each { |output, events| output.multi_receive(events) }
+    outputs_events.each do |output, events|
+      with_pipeline_rescue(events) do
+        output.multi_receive(events)
+      end
+    end
+  end
+
+  def with_pipeline_rescue(event_or_events)
+    yield
+  rescue Exception => e
+    logger_opts = {:message => e.message,
+                   :class => e.class.name,
+                   :backtrace => e.backtrace}
+    logger_message = "Exception in pipelineworker!"
```

Comment:
by default logstash only log stacktraces when logstash is run in debug mode, should we do the same here since stacktraces can get quite noisy ?